### ISSUE TITLE
Spread operator and Union types

### ISSUE THREAD

**Issue by sebdoucet** [Microsoft/TypeScript#6677]
Hi,

As it is right now, it's not possible to do that:

``` Typescript

class Point{
   constructor(public x: number, public y: number){ }
   add(x: number, y: number): void {
     this.x+= x;
     this.y+= y;
  }
}

// Strict data structure
let pointData: [number, number] = [0,0];

// That does not compile
let point = new Point(...pointData);
// That does not compile too
point.add(...pointData);
```

Is there a reason why it's not supported ?

Regards,
Sébastien.

**Comment by sandersn** [comment]
Addressed by proposal #5453.

**Comment by sandersn** [comment]
And by the associated proposal #5296.
